### PR TITLE
Add more tests for dpnp.sum and sum_over_axis_0 extension

### PR DESCRIPTION
**Pull request by AlexanderKalistratov**
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] Have you checked performance impact of proposed changes?
- [x] If this PR is a work in progress, are you filing the PR as a draft?

Some cleanups, more tests for sum_over_axis_0 extension and dpnp.sum

### CODE REVIEW

File: tests/test_mathematical.py [hunk 1]
```diff
@@ -1042,23 +1043,46 @@ def test_sum_empty_out(dtype):
...
 )
 @pytest.mark.parametrize(
     "dtype_out", get_all_dtypes(no_complex=True, no_bool=True)
 )
-def test_sum(shape, dtype_in, dtype_out):
-    a_np = numpy.ones(shape, dtype=dtype_in)
-    a = dpnp.ones(shape, dtype=dtype_in)
-    axes = [None, 0, 1, 2]
+@pytest.mark.parametrize("transpose", [True, False])
+@pytest.mark.parametrize("keepdims", [False])
```
Comment (AlexanderKalistratov) [inline tests/test_mathematical.py:1066]: Setting `keepdims` to `True` results in sum tests to fail.
See details: https://github.com/IntelPython/dpnp/issues/1487

Comment (antonwolfy) [reply tests/test_mathematical.py:1067]: the issue is resolved, could you please add the testing with `keepdims=True`

File: tests/test_mathematical.py [hunk 2]
```diff
@@ -1042,23 +1044,46 @@ def test_sum_empty_out(dtype):
...
+        (0, 6),
+        (10, 1),
+        (1, 10),
+    ],
 )
 @pytest.mark.parametrize(
     "dtype_in", get_all_dtypes(no_complex=True, no_bool=True)
 )
 @pytest.mark.parametrize(
     "dtype_out", get_all_dtypes(no_complex=True, no_bool=True)
```
Comment (antonwolfy) [inline tests/test_mathematical.py:1064]: Can we enable the test for bool and complex dtypes also?

File: tests/test_mathematical.py (hunk 1 above)
Comment (AlexanderKalistratov) [reply tests/test_mathematical.py]: done

File: tests/test_mathematical.py (hunk 2 above)
Comment (AlexanderKalistratov) [reply tests/test_mathematical.py]: done.

Now in total there are about about 1300 tests on sum, which takes in total ~25 seconds.

**Review by antonwolfy — APPROVED** [review]

**Pull request merged**